python/cells.cpp

```diff
@@ -0,0 +1,226 @@
+            [](const regular_schedule_shim& sched){
+                return arb::spike_source_cell{sched.schedule()};}),
+            "schedule"_a, "Construct a spike source cell that generates spikes at regular intervals")
+        .def(pybind11::init<>(
+            [](const explicit_schedule_shim& sched){
+                return arb::spike_source_cell{sched.schedule()};}),
+            "schedule"_a, "Construct a spike source cell that generates spikes at a sequence of user-defined times")
+        .def(pybind11::init<>(
+            [](const poisson_schedule_shim& sched){
+                return arb::spike_source_cell{sched.schedule()};}),
+            "schedule"_a, "Construct a spike source cell that generates spikes at times defined by a Poisson sequence")
+        .def("__repr__", [](const arb::spike_source_cell&){return "<spike_source_cell>";})
+        .def("__str__",  [](const arb::spike_source_cell&){return "<spike_source_cell>";});
+
+    pybind11::class_<arb::benchmark_cell> benchmark_cell(m, "benchmark_cell",
+        "A benchmarking cell, used by Arbor developers to test communication performance.");
+
+    benchmark_cell
+        .def(pybind11::init<>(
+            [](const regular_schedule_shim& sched, double ratio){
+                return arb::benchmark_cell{sched.schedule(), ratio};}),
+            "schedule"_a, "realtime_ratio"_a)
+        .def(pybind11::init<>(
+            [](const explicit_schedule_shim& sched, double ratio){
+                return arb::benchmark_cell{sched.schedule(), ratio};}),
+            "schedule"_a, "realtime_ratio"_a)
+        .def(pybind11::init<>(
+            [](const poisson_schedule_shim& sched, double ratio){
+                return arb::benchmark_cell{sched.schedule(), ratio};}),
+            "schedule"_a, "realtime_ratio"_a)
+        .def("__repr__", [](const arb::spike_source_cell&){return "<benchmark_cell>";})
+        .def("__str__",  [](const arb::spike_source_cell&){return "<benchmark_cell>";});
+
+    pybind11::class_<arb::lif_cell> lif_cell(m, "lif_cell",
+        "A benchmarking cell, used by Arbor developers to test communication performance.");
+
+    lif_cell
+        .def(pybind11::init<>())
+        .def_readwrite("tau_m", &arb::lif_cell::tau_m,  "Membrane potential decaying constant [ms].")
+        .def_readwrite("V_th",  &arb::lif_cell::V_th,   "Firing threshold [mV].")
+        .def_readwrite("C_m",   &arb::lif_cell::C_m,    "Membrane capacitance [pF].")
+        .def_readwrite("E_L",   &arb::lif_cell::E_L,    "Resting potential [mV].")
+        .def_readwrite("V_m",   &arb::lif_cell::V_m,    "Initial value of the Membrane potential [mV].")
+        .def_readwrite("t_ref", &arb::lif_cell::t_ref,  "Refractory period [ms].")
+        .def_readwrite("V_reset", &arb::lif_cell::V_reset, "Reset potential [mV].")
+        .def("__repr__", [](const arb::lif_cell&){return "<lif_cell>";})
+        .def("__str__",  [](const arb::lif_cell&){return "<lif_cell>";});
```

Comment:
- Add description in constructor. 
- Add `arbor: ` in return string for `__str__` and `__repr__`
- Do we want/need defaults for the parameters?

Reply:
Do we need comments for the default constructor, which does not do anything noteworthy.

It uses the defaults for the underlying C++ `lif_cell`.
I have updated the `__str__` and `__repr__` to print out the parameter values:
```python
import arbor
print(arbor.lif_cell()) 
```
generates:
```
<arbor.lif_cell: tau_m 10, V_th 10, C_m 20, E_L 0, V_m 0, t_ref 2, V_reset 0>
```

Reply:
I would add a comment to the default constructor (to keep consistency).

Reply:
What would the message be? The user can see that there is a default constructor by calling `help(arbor.lif_cell)`. Printing a message like `"Construct a lif cell"` is quite redundant to me.

Are we being consistent with other default constructors in the library?

Reply:
So far, I always added a description. We can add this check to issue #782 Python wrapper: uniform naming and printing of types

Reply:
This is a different issue to the printing ticket IMO (it is a documentation issue).

Some types in the wrapper at the moment have a help message for the default constructor, but many don't.

Below is an example of the `help` output for `explicit_schedule`. In this case it is reasonable, to disambiguate from the other constructor, and because it is worth mentioning that the default is an empty schedule.

```
 |  __init__(...)
 |      __init__(*args, **kwargs)
 |      Overloaded function.
 |      
 |      1. __init__(self: arbor.explicit_schedule) -> None
 |      
 |      Construct an empty explicit schedule.
 |      
 |      
 |      2. __init__(self: arbor.explicit_schedule, times: List[float]) -> None
 |      
 |      Construct an explicit schedule with argument:
 |        times: A list of times (in ms, default []).
```

Reply:
Ok, sure. We can only add a description if worth mentioning.